### DISCUSSION TITLE
Dependencies Behaviour Twin & Knowledge Agents Kits

### DISCUSSION THREAD

**Discussion by drcgjung** [eclipse-tractusx/sig-release#203] [Ideas]
This is a placeholder for the topics appearing in the dependency session between the BT (RUL, HI, ...) & Agents Kits for release 24.03

- Using Use Case Templates to Describe, Validate & Discover Graph (& Skill) Assets
  - Create Excel Template from Ontology Release
  - Row is concept (class, attribute, relation)
  - Use Case defines a set of roles
  - For each row, the use case defines  a set of provider roles and a set of consumer roles and an "arity" (0|1 optional, 1 required, >1) and (in the end) an access permission.
- Implementing Ping-Pong Skills: 
   - Concept see CX-0067 Ontology Models in Catena-X V1.0.0 and CX-0084 V Federated Queries in Dataspaces 1.1.0 
   - Skill Moves on the Meta-Level from Goal/Service Description to Preconditions/Data Descriptions
   - Skill Moves on the Data-Level from Preconditions/Data to Goal/Services
- Keep Up-To-Date with Streaming Transfer PoC

**Comment by drcgjung** [comment]
# Feature "Easy & Formal Definition of Use Cases"

## Benefit Hypothesis
* Formal Use Case Specific Artifacts like Credentials, Roles, Data Models, Requirements, Policies are required in many places
* Having a simple "source" definition, such as Excel, provides a single-source-of-truth
* Further formal derivates can be added later by providing an extractor logic
* Maintaining and extending use cases becomes easier
* Participants can "instantiate" their roles over multiple use cases to get their personal view on the dataspace

## Beneficiaries
* any use case modeller and artifact maintainer
* standard authors
* participants to multiple use cases

## Acceptance Criteria
* Use Case excel template format including role definition, contract/asset references and taxonomy choice
* Use Case ttl format concept and meta-model
* excel template updater (when ontology changed)
* generator for excel->ttl (when excel changed)
* generator for excel->policies/contractdefinitions

Estimated Story Points:
13 SP

**Comment by drcgjung** [comment]
# Feature "Definition of Goal-Oriented Skills"

## Benefit Hypothesis
* Having skill templates for common dataspace situations will help the use case and application developers to implement their business logic
* Skills typically enwork data (assets) from the preconditions to the final output.
* Assets need to be restricted/found on the meta-level in the reverse direction (from the goal type to the precondition type)
* Having a template which can bring both directions together in a ping-pong style will allow very flexible data processing patterns.

## Beneficiaries
* any use case developer
* any application developer

## Acceptance Criteria
* Necessary extensions to the CX-0084 and CX-0067 standards related to the description of input and output types of assets
* Skill Template 
* Example Skill realising ping-pong behaviour

Estimated Story Points:
13 SP

**Comment by drcgjung** [comment]
Following things would need to be documented:
- Invocation of remote skills
- Aggregating individual results
- Calling the Catalogue Graph as an Asset

**Comment by stephanbcbauer** [comment]
@drcgjung since this discussion is ancient and my assumption is you used this for feature planning in older releases, I close this discussion.
